spyder/widgets/tests/test_editor.py

```diff
@@ -458,29 +483,219 @@ def test_get_current_word(base_editor_bot):
     assert editor.get_current_word() == 'valid_python_word'
 
 
-def test_editor_splitter_init(editor_splitter_bot):
-    """"Test EditorSplitter.__init__."""
-    es = editor_splitter_bot
-    assert es.orientation() == Qt.Horizontal
-    assert es.testAttribute(Qt.WA_DeleteOnClose)
-    assert not es.childrenCollapsible()
-    assert not es.toolbar_list
-    assert not es.menu_list
-    assert es.register_editorstack_cb == es.plugin.register_editorstack
-    assert es.unregister_editorstack_cb == es.plugin.unregister_editorstack
-
-    # No menu actions in parameter call.
-    assert not es.menu_actions
-    # EditorStack adds its own menu actions to the existing actions.
-    assert es.editorstack.menu_actions
-
-    assert isinstance(es.editorstack, EditorStack)
-    es.plugin.register_editorstack.assert_called_with(es.editorstack)
-    es.plugin.unregister_editorstack.assert_not_called()
-    es.plugin.clone_editorstack.assert_not_called()
-
-    assert es.count() == 1
-    assert es.widget(0) == es.editorstack
+class TestEditorSplitter:
+    """Tests for EditorSplitter methods."""
```

Comment:
@dalthviz   Since editor.py contains many classes, I was trying to write unit tests that tested just the function of one class and not so much the interaction between the classes or the GUI parts overall.  I wanted to specify that these tests were over methods of the EditorSplitter class.  Originally, I didn't have the class and I was writing test names such as `test_editorsplitter_split()`, but I think putting them into a Test Class made it a little less verbose and a little clearer.  For example, `test_init` and `test_close` could apply to any of the classes in the module, so this organizes them as tests just for EditorSplitter.

The other thought I had about using the Test Class was that, if editor.py is refactored where each class becomes its own module, then it would be easy to find the corresponding tests and refactor them into their own source as well.

If you prefer that there aren't classes within the test modules, I can change it.  

FYI, the test_init that I was writing for this PR was how I discovered issue #5475.  These methods alter the state of many properties, so that's what I was trying to add tests for.  If a certain state is expected now, then that's what should be expected after any code change is made.

Reply:
>The other thought I had about using the Test Class was that, if editor.py is refactored where each class becomes its own module, then it would be easy to find the corresponding tests and refactor them into their own source as well.

Yes, moving each class to a module is the idea after split-plugins merge, and test will be in different files too

Maybe you could move this test to a new file instead spyder/widgets/tests/test_editorsplitter.py

Reply:
@rlaverde   I've split the tests into a separate file.